app/src/main/java/com/hedvig/app/feature/home/ui/claimstatus/data/ClaimProgressData.kt

```diff
@@ -0,0 +1,104 @@
+                @Composable get() = ContentAlpha.medium
+        }
+
+        object CurrentlyActive : ClaimProgressType() {
+            override val color: Color
+                @Composable get() = MaterialTheme.colors.primary
+            override val contentAlpha: Float
+                @Composable get() = ContentAlpha.high
+        }
+
+        object FutureInactive : ClaimProgressType() {
+            override val color: Color
+                @Composable get() = MaterialTheme.colors.primary
+            override val contentAlpha: Float
+                @Composable get() = ContentAlpha.disabled
+        }
+
+        object Reopened : ClaimProgressType() {
+            override val color: Color
+                @Composable get() = ClaimStatusColors.Progress.reopened
+            override val contentAlpha: Float
+                @Composable get() = ContentAlpha.high
+        }
+    }
+
+    companion object {
+        fun progressItemListFromClaimStatus(claimStatus: HomeQuery.ClaimStatus): List<ClaimProgressData> {
+            val (first, second, third) = progressItemTypeListFromClaimStatus(claimStatus)
+            return listOf(
+                ClaimProgressData("Submitted", first),
+                ClaimProgressData("Being handled", second),
+                ClaimProgressData("Closed", third),
+            )
+        }
+
+        private fun progressItemTypeListFromClaimStatus(
+            claimStatus: HomeQuery.ClaimStatus
+        ): Triple<ClaimProgressType, ClaimProgressType, ClaimProgressType> {
+            return when (claimStatus.status) {
+                ClaimStatus.SUBMITTED -> Triple(
+                    ClaimProgressType.CurrentlyActive,
+                    ClaimProgressType.FutureInactive,
+                    ClaimProgressType.FutureInactive,
+                )
+                ClaimStatus.BEING_HANDLED -> Triple(
+                    ClaimProgressType.PastInactive,
+                    ClaimProgressType.CurrentlyActive,
+                    ClaimProgressType.FutureInactive,
+                )
+                ClaimStatus.CLOSED -> {
+                    when (claimStatus.outcome) {
+                        ClaimOutcome.PAID -> Triple(
+                            ClaimProgressType.Paid,
+                            ClaimProgressType.Paid,
+                            ClaimProgressType.Paid,
+                        )
+                        else -> Triple(
+                            ClaimProgressType.CurrentlyActive,
+                            ClaimProgressType.CurrentlyActive,
+                            ClaimProgressType.CurrentlyActive,
+                        )
+                    }
+                }
+                ClaimStatus.REOPENED -> Triple(
+                    ClaimProgressType.PastInactive,
+                    ClaimProgressType.Reopened,
+                    ClaimProgressType.FutureInactive,
+                )
+                ClaimStatus.UNKNOWN__ -> throw IllegalArgumentException()
```

Comment:
We need to handle `UNKNOWN__` a bit more gracefully, since this will be matched by all old clients that have not updated their apps in the event of a new `ClaimStatus` 🙂